packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -148,7 +148,7 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         error?: any) {
         const newEvent = { ...event };
         if (error !== undefined) {
-            TelemetryLogger.prepareErrorObject(newEvent, error, false);
+            TelemetryLogger.prepareErrorObject(newEvent, error, true /* fetchStack */);
```

Comment:
the underlying process of "generateStack" involved under here recently caused confusion for FFX component team in understanding whether or not the stack they were reading was actually the stack for the error or if is was surrogate from logging.   When people read a stack, they assume this is 'where an error occurred', which is already ambiguous in these runtimes between instantiation and thrown. So we add a third, it could have been surrogate supplied from logging logic.

I would advise against it, and boldly claim it is an anti-pattern in error handling/logging, but if you must do it use attribute name like 'surrogateStack' and not overload 'stack' on the events.

If I implemented your client, I wouldn't let stack be settable on Error/Exception type :)
This way you could log an event in the same way, but you should not be able to throw errors with potential misinformation across boundaries.
If we could test any real clients behave this way it would be a data point their philosophy on the issue is the same.

Reply:
I wonder what caused confusion, given that today (before this change) it feels like that should be no confusion - if there is an error, then stack (if present) comes from error. If there is no error (and there is stack), it's obvious it does not come from error (we do not have one).

But overall, I'm of same opinion - we should not create "fake" errors (equal to event name) and we should not populate stack if error object does not have it - these steps feel like create more confusion, and do not solve any problems.

Reply:
> I wonder what caused confusion, given that today (before this change) it feels like that should be no confusion

See Justin's edit to his comment - this equivalent logic has existed in Bohemia codebase for a long time. Someone complained about an error in the "getStack" function since that was the end of the stack 😣

Reply:
> > I wonder what caused confusion, given that today (before this change) it feels like that should be no confusion
> 
> See Justin's edit to his comment - this equivalent logic has existed in Bohemia codebase for a long time. Someone complained about an error in the "getStack" function since that was the end of the stack 😣

Yep, as Mark describes it is similar logic in office-bohemia. Not specifically this code.
There are a lot of similarities in the 'utility' and 'helper' type areas of the code bases I've noticed.